tools/half-sign.c

```diff
@@ -0,0 +1,117 @@
+#include <unistd.h>
```

Comment:
oh wow, when did we start using C? Where is a doc for this file? what does it do?